client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -43,14 +45,34 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
     const {
       options: { separator, decimal },
     } = useCurrency();
-    const [textValue, setTextValue] = useState(format(value ?? defaultValue));
+    const [textValue, setTextValue] = useState(
+      format(value ?? defaultValue, { minimumFractionDigits: decimalMin })
+    );
+    const isFirstRender = useRef(true);
+
+    const isInputIncomplete = useCallback(
+      (value: string) =>
+        new RegExp(
+          // Checks for a trailing `.` or a `0` (not necessarily immediately)
+          // after a `.`
+          `^\\d*\\.$|\\d*${RegexUtils.escapeChars(decimal)}\\d*0$`
+        ).test(value),
+      [decimal]
+    );
```

Comment:
I don't love this "firstrender" check, but I'm not sure how else it can do what we want -- it needs to update `value` from `defaultValue` on first load, but then keep our internal `textValue` up to date whenever `value` changes externally after that.